Fix string recognition of ro channel map because of representation sa…

nathlacroix opened the pull request:
Fixes #17
The readout channel map was wrongly acquired due to the string representation of the string. 

The pycqed code has been revised and no other instances have been found with similar issues, as all other cases look for parameter values (i.e. float) where repr(value) == str(value). If accessing a string value from the HDF file, one should apply eval on it.

nathlacroix committed:
Fix string recognition of ro channel map because of representation saving of strings

antsr commented:
But `acq_weights_type` is not the only string parameter in the hdf5 files. We have all the pulse types and channel names, for example.

stephlazar commented:
~@antsr that is true, but I think what Nathan did is to fix all instances in PycQED where string~ ~parameters are extracted from HDF5 file, by adding eval. In the future, we should always eval~ ~whenever we extract.~

@antsr raises a very good point. I didn't notice this pull request only changes one line in one file. Someone must correct all placed in the PycQED repository where we extract string parameters from HDF5 file (there are several other places where this is done). I can take care of this.

nathlacroix commented:
All the instances I have found (approx. 300) are either outdated functions in the analysis toolbox or they look for parameters which are numbers, hence eval(value) == value . But maybe I overlooked some of them. We can check that together @stephlazar

Commit:
Removes FluxDecoupler.py, conversion_box_control.py, device_object.py, heterodyne.py.
Adds repr wherever parameters are loaded from Instrument settings and MC in HDF5 file:
   in analysis_toolbox.py and measurement_analysis.py.
Cleans up measurement_analysis.py (deletes unused classes).

stephlazar commented:
@antsr @nathlacroix I made more changes. Please review.

antsr commented on pycqed/analysis/analysis_toolbox.py:
```diff
@@ -344,10 +350,11 @@ def get_qb_channel_map_from_hdf(qb_names, file_path, value_names, h5mode='r+'):
     for qbn in qb_names:
-        qbchs = [str(instr_settings[qbn].attrs['acq_I_channel'])]
-        ro_acq_weight_type = instr_settings[qbn].attrs['acq_weights_type']
+        qbchs = [str(eval(instr_settings[qbn].attrs['acq_I_channel']))]
```
`str` should not be needed anymore

antsr commented on pycqed/analysis/analysis_toolbox.py:
```diff
@@ -317,7 +322,8 @@ def get_param_value_from_file(file_path, instr_name, param_name, h5mode='r+'):
     if instr_name in list(instr_settings.keys()):
         if param_name in list(instr_settings[instr_name].attrs):
-            param_val = float(instr_settings[instr_name].attrs[param_name])
+            param_val = float(eval(instr_settings[instr_name].attrs[
```
`float` should not be needed anymore with eval

antsr commented on pycqed/analysis/measurement_analysis.py:
```diff
@@ -11,58 +10,46 @@
 except ImportError as e:
     if str(e).find('pygsti') >= 0:
-        logging.warning('Could not import pygsti')
+        logger.warning('Could not import pygsti')
```
Not a critical issue, but just to raise awareness, that the logger object is usually (in other modules of PycQED) called `log`. I know it was a problem already before.

antsr commented on pycqed/instrument_drivers/meta_instrument/FluxDecoupler.py:
```diff
@@ -1,136 +0,0 @@
-import numpy as np
```
This class should still be working. I wrote it at some point to store and use the flux decoupling matrix. I would not delete it.

antsr reviewed (changes requested):
thanks for the changes, @stephlazar 
the `FluxDecoupler` class should not be deleted, it should still work and might be useful.
Also, I think that when we do an `eval`, we shouldn't have to do any further casting, e.g. `str(eval(...` and `float(eval(...` and `int(eval(...` should be replaced with just `eval(...`

nathlacroix reviewed (commented):
@stephlazar thanks for the changes and the clean up !
I will address the comments of @antsr asap (this weekend or next week)

Commit:
Removes unnecessary data type conversions after eval of parameters loaded from HDF5 file in
   analysis_toolbox.py and measurement_analysis.py.
Renames logger to log in measurement_analysis.py.
Adds back FluxDecoupler.py to instrument_drivers/meta_instrument.

stephlazar commented:
@nathlacroix I already did :)

stephlazar reviewed (dismissed):
Now it looks ok

Commit:
Fixes bug in get_qb_channel_map_from_hdf, where acq_I_channel and acq_Q_channel were
    int instead of str.

stephlazar commented:
I have now tested a few standard analyses that we run and fixed a small bug.
Analyses tested: Rabi with SSB weights, Rabi with classifier and active reset, Ramsey (SSB), QScale (SSB), T1 (SSB), Flux pulse scope, Chevron. 

The pull request can be merged from my point of view.
@antsr @nathlacroix
